manim/cli/project/commands.py

```diff
@@ -0,0 +1,308 @@
+
+new  -  The new command group has 2 commands. these commands handle project creation
+        and scene creation or insertion
+
+        project -   The project subcommand is used for project creation
+                    This command is similar to init but different in a way
+                    that it asks for project_name and template_name.
+                    init command initializes a new project in the current dir
+                    while project command creates a new folder and creates the
+                    project there
+
+        scene   -   The scene subcommand command is used for inserting new scenes.
+                    scene command can create new files and insert new scenes in there.
+                    It can also insert new scenes in an already existing python file.
+
+How To:
+    create project
+        manim new project
+
+        manim new project <project name>
+
+        manim new project <project name> <template name>
+
+    Init project
+        manim init
+
+    add or insert scene
+        manim scene <scene name> : add <scene name> in main.py
+
+        manim scene <scene name> <file name> : add scene in <file name>.
+                                                if <file name> does not exist creates new <file name>
+                                                and insert <scene name>.
+"""
+
+import configparser
+from pathlib import Path
+from shutil import copyfile
+
+import click
+
+from ... import console
+from ...constants import CONTEXT_SETTINGS, EPILOG, QUALITIES
+
+# when -d flag is passed in 'project new' command CFG_DEFAULTS is passed to update_cfg() function
+CFG_DEFAULTS = {
+    "frame_rate": 30,
+    "background_color": "BLACK",
+    "background_opacity": 1,
+    "scene_names": "Default",
+    "resolution": (854, 480),
+}
+
+
+def get_template_names():
+    """Gets template names from the templates directory
+
+    returns:
+            list of names of python files in templates direcotory
+    """
+    template_path = Path.resolve(Path(__file__).parent / "templates")
+    return [template_name.stem for template_name in template_path.glob("*.py")]
+
+
+"""\
+    utility functions are helper functions that provide some core functionality
+    to do project management
+
+    functions:
+    1 - update_cfg:
+            used for updating manim.cfg file when -d flag is not provided in
+            the 'manim new project' command
+
+    2 - copy_template_files:
+            copies template file upon successful project creation.
+            it is used in both 'manim init' and 'manim project new' commands
+
+    3 - select_resolution:
+            when 'manim project new' command is issued without the -d flag.
+            this function does conversions of 'QUALITIES' constant in to a
+            workable format for the click.Choice prompt. after the prompt it
+            returns a tuple containing pixel width and height based on the
+            result from the prompt.
+"""
```

Comment:
This content should go under each of the respective helper functions as part of their docstrings. Also, remember capitalization/punctuation.

Reply:
okay thanks again for the review. i will start working on it

Reply:
done